### DISCUSSION TITLE
Task name quite restrictive (globally unique and relatively short name)

### DISCUSSION THREAD

**Discussion by richtier** [celery/django-celery-beat#474] [Ideas]
### Summary:

* Celery Version: 4.4.7
* Celery-Beat Version: 2.1.0 

### Detailed information

The task name must be unique as the code has `unique=True`

This is a bit a problem for my usecase:

- multiple django apps (entirely different codebase) all talk to the same queue
- a different django app (again entirely different codebase) pushes periodic tasks to the queue
- the tasks have "queue" set, so "app_a" gets tasks on "app_a" queue.

The problem is if two apps both have "debug task" function then what do we call the function? has to be unique. Could prefix with the queue name, but I dont think that is scalable  because the max_length of name is 200. That problem is compounded when we consider a multiple tasks can be created for the same function with different args. Do we suffix the args in the name too.

Simplest solution I think it to change name to `TextField` so no max length.

**Comment by auvipy** [comment]
I have to wait for more community input

**Comment by melicheradam** [comment]
Hi @auvipy 

Im also facing problems with the restrictions put on the task name. Is there a particular reason for it to be unique? 

For example, I want to generate two notifications for an event happening in my app, one for start, one for end. These notifications will be scheduled via a periodic task with ClockedSchedule. Now it is reasonable to set the name the same, but I cannot. Not a big problem, but I cannot see the reasoning behind the restrictions. 

Maybe a better approach would be to create a new field called task_group or something, if you want to index that field and make it unique, grouping multiple tasks with the same context together.

**Comment by oskar-s9s** [comment]
What's the point of task name being unique if we have ID PK?
I'd like to reuse task name - multiple tasks and cannot now.
I have to generate random string which makes this name non-displayable in app.
